### DISCUSSION TITLE
How to limit the number of traces

### DISCUSSION THREAD

**Discussion by secsilm** [langfuse/langfuse#1785] [Support]
If you keep all data all the time, it will consume a large amount of hard disk space. Is there an option to set a TTL expiration time, such as only keeping data from the last month?

**Comment by marcklingen** [comment]
If you self-host Langfuse, you need to manage data retention yourself. This is a planned feature for a future release of langfuse. If you delete data from Postgres, make sure to delete from traces, observations, and events tables

**Comment by TheErk** [comment]
Any news on that subject ?
Does LF v3 comes with a feature which makes it possible to limit the number of traces?

**Reply by marcklingen** [reply]
We have added a data retention setting but this is more geared towards compliance requirements

**Reply by TheErk** [reply]
Where is this settings? Is it available on self-hosted instance?

**Reply by marcklingen** [reply]
It is part of the commercially licensed features as it is usually only required for compliance in larger organizations. You can run the scripts here on your own db if you want to use it in Langfuse OSS: https://github.com/orgs/langfuse/discussions/2268

**Reply by TheErk** [reply]
This is nice. Then I guess adding TTL would be possible as well cf: https://github.com/orgs/langfuse/discussions/2268#discussioncomment-12069669
